storage/storage.go

```diff
@@ -41,3 +41,19 @@ type STHWriter interface {
 type RootsWriter interface {
 	WriteRoots(ctx context.Context, l *ctlog.Log, roots []*x509.Certificate, receivedAt time.Time) error
 }
+
+// RootSetID uniquely identifies a specific set of certificates, regardless of their order.
+type RootSetID string
```

Comment:
As a string, it can be used as the key in a map. This is helpful since that's exactly what it is - a key for referring to a set of certificates. If I make it a []byte, I have to cast it to string whenever I used it with a map. It is also trivially comparable with other RootSetIDs as a string. However, there aren't any significant obstacles to making it a []byte so I'd be fine with changing it, if you've got strong reasons for doing so? The only one I can think of is that it's not really a printable string, so []byte is semantically more accurate.

Reply:
I'm relaxed, I've been going back and forth on this in my head. It looks like the only map place it's used atm is for testing, and so that shouldn't be the sole reason for it to be a string, but I also don't have strong reasons why it should be []byte, so whichever is fine.  I think in my head we'd be calculating it by doing a bunch of hashing, so that felt bytey, but then having the ID as []byte might cause problems if we want to use it as part of a primary key in a DB, depending on the db type.  So author's choice - I leave it up to you :)